src/components/form/HdRange.vue

```diff
@@ -0,0 +1,320 @@
+<template>
+  <div
+    :class="fieldClasses"
+    class="range field"
+  >
+    <input type="range"
+      ref="input"
+      v-model="currentValue"
+      :id="name"
+      :name="name"
+      :required="required"
+      :disabled="disabled"
+      :min="minValue"
+      :max="maxValue"
+      :step="rangeStep"
+      @input="updateRangeDecoration"
+      @change="handleChange"
+      @focus="handleFocus"
+      @blur="handleBlur"
+    >
+    <div class="range__thumb" ref="thumb">
+      <div class="range__thumb__bullet">
+      </div>
+    </div>
+    <div class="range__decoration">
+      <div class="range__progress" ref="progress">
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+  name: 'HdRange',
+  props: {
+    name: {
+      type: String,
+      default: 'range',
+    },
+    required: {
+      type: Boolean,
+      default: false,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    minValue: {
+      type: Number,
+      default: 0,
+    },
+    maxValue: {
+      type: Number,
+      default: 100,
+    },
+    rangeStep: {
+      type: Number,
+      default: 1,
+    },
+    isLegacy: {
+      type: Boolean,
+      default: false,
+    },
+    value: {
+      type: Number,
+      default: 0,
+    },
+  },
+  data() {
+    return {
+      isActive: null,
+      currentValue: this.value,
+    };
+  },
+  computed: {
+    fieldClasses() {
+      return {
+        'field--active': this.isActive,
+        'field--disabled': this.disabled,
+      };
+    },
+  },
+  watch: {
+    minValue() {
+      this.updateRangeDecoration();
+    },
+    maxValue() {
+      this.updateRangeDecoration();
+    },
+    rangeStep() {
+      this.updateRangeDecoration();
+    },
+    value() {
+      this.currentValue = this.value;
+      this.updateRangeDecoration();
+    },
```

Comment:
Here we can make use of the modal (that's how we won't need to handle the change/input events ourselves):

```js
value() {
  this.currentValue = this.value;
},
currentValue() {
  this.updateRangeDecoration();
},
```

Reply:
🚀